custom_components/versatile_thermostat/thermostat_switch.py

```diff
@@ -27,7 +27,7 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class ThermostatOverSwitch(BaseThermostat):
+class ThermostatOverSwitch(BaseThermostat[UnderlyingSwitch]):
```

Comment:
In case you were not familiar (sorry if redundant), this is the expression that assigns a concrete type — here, `UnderlyingSwitch` — to the generic `T` TypeVar defined in `base_thermostat.py`.